benchmarks/mongo-es-speed/prepareEvents.js

```diff
@@ -34,18 +36,17 @@ export default function (eventsCount, reportObj) {
     let processedEvents = 0;
 
     return Array.from(new Array(eventsCount)).reduce(
-        acc => acc.then(() =>
-            store.saveEvent({
-                type: TYPES[Math.floor(Math.random() * TYPES.length)],
-                aggregateId: `GUID${numericRandom(20)}`,
-                payload: buildPayload(),
-                timestamp: Date.now()
-            }).then(() => ((++processedEvents % 5000 === 0)
-                ? (reportObj.value += 5000)
-                : null
-            ))
-        ),
+        acc =>
+            acc.then(() =>
+                store
```

Comment:
Stairs-alike code. `store` does not produce promise, which is chained in bottom code, but `store.saveEvents` does - so it should be monolith construction.